src/game-logic/reducers/updatePeer.js

```diff
@@ -1,18 +1,49 @@
+/**
+ * @typedef {import('../../components/Farmhand/Farmhand').default} Farmhand
+ * @typedef {import('../../components/Farmhand/Farmhand').farmhand.state} farmhand.state
+ * @typedef {import('../../index').farmhand.peerMetadata} farmhand.peerMetadata
+ */
 import { MAX_LATEST_PEER_MESSAGES } from '../../constants'
+import { NEW_COW_OFFERED_FOR_TRADE } from '../../templates'
+import { dialogView } from '../../enums'
+
+import { showNotification } from './showNotification'
 
 /**
  * @param {farmhand.state} state
+ * @param {Farmhand} farmhand
+ * @param {farmhand.peerMetadata} peerMetadata
  * @param {string} peerId The peer to update
- * @param {Object} state
  * @returns {farmhand.state}
  */
-export const updatePeer = (state, peerId, peerState) => {
+export const updatePeer = (state, farmhand, peerMetadata, peerId) => {
   const peers = { ...state.peers }
-  peers[peerId] = peerState
+
+  const previousPeerMetadata = peers[peerId]
+
+  const previousCowOfferedId = previousPeerMetadata?.cowOfferedForTrade?.id
+  const newCowOfferedId = peerMetadata.cowOfferedForTrade?.id
+
+  const isCowNewlyBeingOfferedForTrade =
+    newCowOfferedId &&
+    previousCowOfferedId !== peerMetadata.cowOfferedForTrade?.id
```

Comment:
```suggestion
    previousCowOfferedId !== newCowOfferedId
```

also, is the ID ever 0? if so this check will fail when it is. does it even need the first part or can you just check if the two id's are mismatch?

Reply:
Cow `id`s are strings (UUIDs, specifically), so a valid ID should never be falsy.

> does it even need the first part or can you just check if the two id's are mismatch?

Yes, the `newCowOfferedId &&` check prevents a notification from being displayed when an offering player rescinds their cow without a new one being offered in its place.

I realize now that the intent here isn't super obvious, so I'll add some test coverage for it!

Reply:
Tests added in https://github.com/jeremyckahn/farmhand/commit/f6ebe36b0b7d7d4610f5030558a00be080e87837. I also made some test stub and typing improvements in this commit.

Reply:
ah good to know about the id. it's just semantics but it might be nice to name the property "uid" whenever this is the case instead of just "id" because i strongly associate "id" to integer values, but i'm sure you have a typedef that also could have told me this 😄. 

any thoughts on my suggestion that i started this comment thread with? it seems like you're doing the same check twice when you could just compare your previous to next variables

Reply:
> ah good to know about the id. it's just semantics but it might be nice to name the property "uid" whenever this is the case instead of just "id" because i strongly associate "id" to integer values, but i'm sure you have a typedef that also could have told me this 😄.

I think it's a good idea, but it would involve having to update users' save data (because these IDs are persisted there). We have infrastructure to do that, but I think it's worth avoiding unless absolutely necessary because it adds a lot of risk to deployments. 😕 I think in this case we're better off leaving it for now.

> any thoughts on my suggestion that i started this comment thread with? it seems like you're doing the same check twice when you could just compare your previous to next variables

Whoops I totally overlooked that, sorry! 😅 This is fixed in https://github.com/jeremyckahn/farmhand/commit/cb34b1d1121e30a73193379ce20bcd665eb61e29. Thanks for re-raising that!